src/components/TodosProvider/TodosProvider.tsx

```diff
@@ -0,0 +1,233 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from '../../api/todos';
+
+import { ErrorMessage } from '../../types/ErrorMessage';
+import { FilterTypes } from '../../types/FIlterTypes';
+import { Todo, UpdateData } from '../../types/Todo';
+import { getFilteredTodos } from '../../utils/getFilteredTodos';
+
+const USER_ID = 6359;
+
+export type ContextType = {
+  todos: Todo[],
+  tempTodo: Todo | null,
+  processedTodos: Todo[],
+  activeTodos: Todo[],
+  completedTodos: Todo[],
+  hasError: boolean,
+  errorMessage: ErrorMessage,
+  inputDisabled: boolean,
+  handleAddTodo: (todoTitle: string) => void,
+  handleDelete: (todo: Todo) => void,
+  handleUpdateTodo: (todo: Todo, fieldsToUpdate: UpdateData) => void,
+  handleToggleAll: () => void,
+  changeHasError: (value: boolean) => void,
+  deleteAllCompleted: () => void,
+  USER_ID: number,
+};
+
+export const TodosContext = React.createContext<ContextType>({
+  todos: [],
+  tempTodo: null,
+  processedTodos: [],
+  activeTodos: [],
+  completedTodos: [],
+  hasError: false,
+  errorMessage: ErrorMessage.NONE,
+  inputDisabled: false,
+  handleAddTodo: () => { },
+  handleDelete: () => { },
+  handleUpdateTodo: () => { },
+  handleToggleAll: () => { },
+  changeHasError: () => { },
+  deleteAllCompleted: () => { },
+  USER_ID,
+});
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const TodosProvider: React.FC<any> = (
```

Comment:
it is not a really good idea to store all handlers in context, for this it is okay to leave them here, but for the future: 
- store global data
- functions which you use more than in 1 place